internal/queryexecutor/execute_conditions_test.go

```diff
@@ -0,0 +1,553 @@
+					conditions: map[string]interface{}{
+						"$gte": "a",
+						"$lte": "f",
+					},
+				},
+				"attr2": {
+					valueType: types.Type_BOOLEAN,
+					conditions: map[string]interface{}{
+						"$eq": true,
+					},
+				},
+				"attr3": {
+					valueType: types.Type_STRING,
+					conditions: map[string]interface{}{
+						"$gte": "a5",
+					},
+				},
+			},
+			expectedKeys: nil,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+
+		t.Run(tt.name, func(t *testing.T) {
+			keys, err := env.e.executeAND(dbName, tt.attrsConds)
+			require.NoError(t, err)
+			require.Equal(t, tt.expectedKeys, keys)
+		})
+	}
+}
+
+func TestExecuteOR(t *testing.T) {
+	env := newTestEnv(t)
+	defer env.cleanup()
+
+	dbName := "testdb"
+	setupDBForTestingExecutes(t, env.e.db, dbName)
+
+	tests := []struct {
+		name         string
+		attrsConds   attributeToConditions
+		expectedKeys map[string]bool
+	}{
+		{
+			name: "non-empty result",
+			attrsConds: attributeToConditions{
+				"attr1": {
+					valueType: types.Type_STRING,
+					conditions: map[string]interface{}{
+						"$gte": "b",
+						"$lt":  "c",
+					},
+				},
+				"attr2": {
+					valueType: types.Type_BOOLEAN,
+					conditions: map[string]interface{}{
+						"$eq": true,
+					},
+				},
+				"attr3": {
+					valueType: types.Type_STRING,
+					conditions: map[string]interface{}{
+						"$gte": "a1",
+						"$lt":  "a2",
+					},
+				},
+			},
+			expectedKeys: map[string]bool{
+				"key4": true,
+				"key5": true,
+				"key1": true,
+				"key2": true,
+				"key3": true,
+			},
+		},
+		{
+			name: "more matches: non-empty result",
+			attrsConds: attributeToConditions{
+				"attr1": {
+					valueType: types.Type_STRING,
+					conditions: map[string]interface{}{
+						"$gte": "a",
+						"$lte": "c",
+					},
+				},
+				"attr2": {
+					valueType: types.Type_BOOLEAN,
+					conditions: map[string]interface{}{
+						"$eq": false,
+					},
+				},
+				"attr3": {
+					valueType: types.Type_STRING,
+					conditions: map[string]interface{}{
+						"$gte": "a2",
+					},
+				},
+			},
+			expectedKeys: map[string]bool{
+				"key1":  true,
+				"key2":  true,
+				"key3":  true,
+				"key4":  true,
+				"key5":  true,
+				"key6":  true,
+				"key7":  true,
+				"key8":  true,
+				"key9":  true,
+				"key11": true,
+				"key21": true,
+			},
+		},
+		{
+			name: "empty results",
+			attrsConds: attributeToConditions{
+				"attr1": {
+					valueType: types.Type_STRING,
+					conditions: map[string]interface{}{
+						"$gte": "zaa",
+					},
+				},
+				"attr3": {
+					valueType: types.Type_STRING,
+					conditions: map[string]interface{}{
+						"$gte": "a3",
+					},
+				},
+			},
+			expectedKeys: nil,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+
+		t.Run(tt.name, func(t *testing.T) {
+			keys, err := env.e.executeOR(dbName, tt.attrsConds)
+			require.NoError(t, err)
+			require.Equal(t, tt.expectedKeys, keys)
+		})
+	}
+}
+
+func TestExecute(t *testing.T) {
```

Comment:
I am missing a test case for  `gt X` and `lt Y` where `X>Y`, expecting empty result

Reply:
Done